app/page.tsx

```diff
@@ -1,21 +1,109 @@
 "use client";
+import useSWR from "swr";
 import Hero from "./components/Hero/Hero";
 import MovieSection from "./components/MovieSection/MovieSection";
-import { movieLists, tabs } from "./mockdata";
+import { movieLists } from "./mockdata";
+import { apiKey, baseApiUrl, fetcher, imgUrl } from "./api";
+import { useState } from "react";
 
 export default function Home() {
-  const handleClick = () => {
-    return alert("Image Clicked");
-  };
+  const [timeWindow, setTimeWindow] = useState<"day" | "week">("day");
+  const { data: popularData, error: popularError } = useSWR(
+    `${baseApiUrl}/3/movie/popular?api_key=${apiKey}`,
+    fetcher
+  );
+  const { data: trendingData, error: trendingError } = useSWR(
+    `${baseApiUrl}/3/trending/movie/${timeWindow}?api_key=945747413f8ed82865430ae895bdf874`,
+    fetcher
+  );
+
+  if (!popularData && !popularError) {
+    return <div>...Loading...</div>;
+  }
+
+  if (popularError) {
+    return <div>Error: {popularError.message || "An error occurred"}</div>;
+  }
+  const popularMovies = popularData.results.map((movie: any) => {
+    return {
+      image: {
+        imgSrc: `${imgUrl}${movie.poster_path}`,
+        altText: "Image Not Found",
+      },
+      title: movie.title,
+      date: movie.release_date,
+      rating: Math.floor(movie.vote_average * 10),
+      MenuPopoverLinks: [
+        {
+          text: "Option 1",
+          url: "https://www.google.com",
+        },
+        {
+          text: "Option 2",
+          url: "https://www.google.com",
+        },
+        {
+          text: "Option 3",
+          url: "https://www.google.com",
+        },
+        {
+          text: "Option 4",
+          url: "https://www.google.com",
+        },
+      ],
+    };
+  });
+
+  if (!trendingData && !trendingError) {
+    return <div>...Loading...</div>;
+  }
+
+  if (trendingError) {
+    return <div>Error: {trendingError.message || "An error occurred"}</div>;
+  }
+  const trendingMovies = trendingData.results.map((movie: any) => {
```

Comment:
@azmat239 
Since we are doing the same thing twice, can we create a fumction to do this action and call the function twice?
Ex.
const getFormattedMovies = (movies)=> movies.map((movie: any) => {
return {
movieImage: {
imgSrc: ${imgUrl}${movie.poster_path},
altText: "Image Not Found",
url: "/",
},
title: movie.title,
date: movie.release_date,
rating: Math.floor(movie.vote_average * 10),
popupMenuLinks: popupMenuLinks,
};
});

then
const trendingMovies = getFormattedMovies(trendingData.results).
const popularMovies = getFormattedMovies(popularData.results).